pkg/domain/infra/tunnel/containers.go

```diff
@@ -381,7 +381,29 @@ func (ic *ContainerEngine) Config(_ context.Context) (*config.Config, error) {
 }
 
 func (ic *ContainerEngine) ContainerPort(ctx context.Context, nameOrId string, options entities.ContainerPortOptions) ([]*entities.ContainerPortReport, error) {
-	return nil, errors.New("not implemented")
+	var (
+		reports    []*entities.ContainerPortReport
+		namesOrIds []string
+	)
+	if len(nameOrId) > 0 {
+		namesOrIds = append(namesOrIds, nameOrId)
+	}
+	ctrs, err := getContainersByContext(ic.ClientCxt, options.All, namesOrIds)
+	if err != nil {
+		return nil, err
+	}
+	for _, con := range ctrs {
+		if con.State != define.ContainerStateRunning.String() {
+			continue
+		}
```

Comment:
Not sure, but I don't think we should add them to paused containers either.

Reply:
This code is ignoreing all non running containers including paused containers, I believe?

Reply:
should be

Reply:
DOH, don't know what I was reading earlier, ignore that comment.